tests/rbd/rbd_delete_protected_snapshot_krbd.py

```diff
@@ -0,0 +1,97 @@
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.krbd_io_handler import krbd_io_handler
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def rbd_delete_protect_snap(rbd, pool_type, **kw):
+    """
+    This module on krbd client - Performs deletion of protected snapshot on a RBD image
+    Args:
+        rbd: RBD object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: test data
+    """
+    try:
+        pool = kw["config"][pool_type]["pool"]
+        image = kw["config"][pool_type]["image"]
+        snap = kw["config"][pool_type].get("snap", f"{pool}_{image}_snap")
+        clone_prefix = kw["config"][pool_type].get("clone", f"{pool}_{image}_clone")
+        size = kw["config"][pool_type]["size"]
+        snap_name = f"{pool}/{image}@{snap}"
+        image_spec = f"{pool}/{image}"
+
+        # Run fio and krbd map operations on image
+        kw.get("config")["image_spec"] = [image_spec]
+        kw.get("config")["size"] = size
+        kw["rbd_obj"] = rbd
+        krbd_io_handler(**kw)
+
+        if rbd.snap_create(pool, image, snap):
+            log.error(f"Creation of snapshot {snap} failed")
+            return 1
+
+        if rbd.protect_snapshot(snap_name):
+            log.error(f"Snapshot protect failed for {snap_name}")
+            return 1
+
+        log.info("Create multiple clones from the snap image")
+        for i in range(1, 11):
+            clone_name = f"{clone_prefix}_{i}"
+            if rbd.create_clone(snap_name, pool, clone_name):
+                log.error(f"Creation of clone {clone_name} failed")
+                return 1
+            else:
+                log.info(f"Clone {clone_name} created successfully")
+
+        if rbd.snap_remove(pool, image, snap):
+            log.error(f"Removal of protected {snap} failed")
+            return 0
+        log.info(f"{snap} snapshot removed even after the snapshot is protected")
+        return 1
```

Comment:
Since we have a negative TC here, we need to reverse the info and error here

Failure of snap removal is an info and being able to delete image is an ERROR from this TC perspective here, otherwise report portal might get confused along with others looking at the logs.